src/engine/enginebuffer.h

```diff
@@ -346,11 +347,12 @@ class EngineBuffer : public EngineObject {
 
     // Object used to perform waveform scaling (sample rate conversion)
     EngineBufferScale* m_pScale;
+    FRIEND_TEST(EngineBufferTest, SlowRubberBand);
     // Object used for linear interpolation scaling of the audio
-    EngineBufferScaleLinear* m_pScaleLinear;
+    EngineBufferScale* m_pScaleLinear;
     // Object used for pitch-indep time stretch (key lock) scaling of the audio
-    EngineBufferScaleST* m_pScaleST;
-    EngineBufferScaleRubberBand* m_pScaleRB;
+    EngineBufferScale* m_pScaleST;
+    EngineBufferScale* m_pScaleRB;
```

Comment:
Can you point out where? I have only found setScalerForTest() which requires EngineBufferScale* m_pScaleLinear, but not the other two. 
I think the original scaler linear is leaking once you replace it with a mock. 
I think it is best to introduce a new pointer which points to the current scratching scaler, in a same way we have it for the keylock scaler.  This should probably avoid all downcasts.